server/src/main/scala/actors/persistent/nebo/NeboArrivalActor.scala

```diff
@@ -0,0 +1,89 @@
+package actors.persistent.nebo
+
+import actors.persistent.nebo.NeboArrivalActor.getRedListPassengerFlightKey
+import actors.persistent.staffing.GetState
+import actors.persistent.{RecoveryActorLike, Sizes}
+import actors.serializers.NeboArrivalMessageConversion
+import akka.actor.Props
+import akka.persistence.{Recovery, SaveSnapshotSuccess, SnapshotSelectionCriteria}
+import drt.shared.CrunchApi.MillisSinceEpoch
+import drt.shared.{NeboArrivals, RedListPassengers, SDateLike}
+import org.slf4j.{Logger, LoggerFactory}
+import scalapb.GeneratedMessage
+import server.protobuf.messages.NeboPassengersMessage.NeboArrivalMessages
+
+object NeboArrivalActor {
+  def props(redListPassengers: RedListPassengers, now: () => SDateLike): Props =
+    Props(new NeboArrivalActor(redListPassengers, now, Option(now().millisSinceEpoch)))
+
+  def getRedListPassengerFlightKey(redListPassengers: RedListPassengers): String = {
+    s"${redListPassengers.flightCode.toLowerCase}-${redListPassengers.scheduled.getFullYear()}-${redListPassengers.scheduled.getMonth()}-${redListPassengers.scheduled.getDate()}-${redListPassengers.scheduled.getHours()}-${redListPassengers.scheduled.getMinutes()}"
+  }
+}
+
+class NeboArrivalActor(redListPassengers: RedListPassengers,
+                       val now: () => SDateLike,
+                       maybePointInTime: Option[MillisSinceEpoch]) extends RecoveryActorLike {
+
+  override val log: Logger = LoggerFactory.getLogger(f"$getClass")
+  override val recoveryStartMillis: MillisSinceEpoch = now().millisSinceEpoch
+  override val snapshotBytesThreshold: Int = Sizes.oneMegaByte
+  override val maybeSnapshotInterval: Option[Int] = Option(maxSnapshotInterval)
+  private val maxSnapshotInterval = 250
+
+  var state: NeboArrivals = NeboArrivals.empty
+
+  override def recovery: Recovery = maybePointInTime match {
+    case None =>
+      Recovery(SnapshotSelectionCriteria(Long.MaxValue, maxTimestamp = Long.MaxValue, 0L, 0L))
+    case Some(pointInTime) =>
+      val criteria = SnapshotSelectionCriteria(maxTimestamp = pointInTime)
+      Recovery(fromSnapshot = criteria, replayMax = maxSnapshotInterval)
+  }
+
+  override def processRecoveryMessage: PartialFunction[Any, Unit] = {
+    case neboArrivalMessages: NeboArrivalMessages =>
+      val neboArrivals: NeboArrivals = NeboArrivalMessageConversion.messageToNeboArrivalMessages(neboArrivalMessages)
+      val keys = state.arrivalRedListPassengers.keys ++ neboArrivals.arrivalRedListPassengers.keys
```

Comment:
You don't need to iterate over all the keys, only the new ones with a `foldLeft` like this:
```
      neboArrivals.arrivalRedListPassengers.foldLeft(state.arrivalRedListPassengers) {
        case (acc, (key, nextPax)) => acc.updated(key, acc.getOrElse(key, Set()) ++ nextPax)
      }
```